apps/src/music/blockly/extensions.js

```diff
@@ -111,3 +115,34 @@ export const playMultiMutator = {
     this.updateShape_(this.extraSoundInputCount_ - 1);
   },
 };
+
+/*
+ * Extension for the effects field, which replaces the input_dummy in the block
+ * with a dynamic field_dropdown and updates the effect value dropdown
+ * based on the selected effect name.
+ */
+export const effectsFieldExtension = function () {
+  // Set the initial state when the block gets created
+  const thisBlock = this;
+  const valuesDropdown = new Blockly.FieldDropdown(function () {
+    return FIELD_EFFECTS_OPTIONS[thisBlock.getFieldValue(FIELD_EFFECTS_NAME)];
+  });
+  valuesDropdown.setValue(DEFAULT_EFFECT_VALUE);
+  this.getInput(FIELD_EFFECTS_VALUE).appendField(
+    valuesDropdown,
+    FIELD_EFFECTS_VALUE
+  );
+
+  // Set up handler to update the effect value when the effect name changes
+  const effectNameField = this.getField(FIELD_EFFECTS_NAME);
+  const baseHandler = effectNameField.onItemSelected_;
+  effectNameField.onItemSelected_ = (menu, menuItem) => {
+    // Update the effect value dropdown's options to match the newly selected effect
+    const newNameValue = menuItem.opt_value;
```

Comment:
Yeah, field dropdown options have a label and a value, and then the block has an effect name and effect value dropdown >.> 

maybe 'selectedEffectName' would be clearer? cause it's the effect name selected in the effect name dropdown. (it is technically the value of the option, but yeah it's confusing since value is overloaded)

Also just realized I'm calling it effectNameField here instead of fieldEffectName like it is everywhere else. Lemme update the naming to both and hopefully it'll be more clear! Thanks for the feedback!